### PR TITLE
IT'S CHAINSAW TIME!! Adds the chainsaw arm implant

### PR DESCRIPTION
**Pull request by Enbyy**
## About The Pull Request

It's CHAINSAW TIME!! Adds a new traitor item, the chainsaw implant. It functions like the laser arm implant, but instead of a hidden laser in your arm you can take out, it's a one handed chainsaw. 
image
The chainsaw implant is exclusive to Engineer, Chief Engineer, Botanist, Assistant, and Tech Support. It costs 12 tc and requires 300 rep to purchase. You can use a screw driver to take the implant out of the autosurgeon, then screwdriver it again to swap arms. 

## Why It's Good For The Game
YOU CAN BECOME CHAINSAW MAN FROM FAMOUS ANIME CHAINSAW MAN.

## Changelog

:cl:
add: Chainsaw arm implant
/:cl:

### CODE REVIEW

File: orbstation/code/antagonists/traitor/items/intergrated_chainsaw.dm [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	force = 24
+	throwforce = 0
+	throw_range = 0
+	throw_speed = 0
+	sharpness = SHARP_EDGED
+	attack_verb_continuous = list("saws", "tears", "lacerates", "cuts", "chops", "dices")
+	attack_verb_simple = list("saw", "tear", "lacerate", "cut", "chop", "dice")
+	hitsound = 'sound/weapons/chainsawhit.ogg'
+	tool_behaviour = TOOL_SAW
+	toolspeed = 1
```
Comment (lizardqueenlexi) [inline orbstation/code/antagonists/traitor/items/intergrated_chainsaw.dm:43]:
```suggestion
/obj/item/chainsaw/mounted_chainsaw/implanted
	name = "implanted chainsaw"
	desc = "A chainsaw that has extended from your arm."
	item_flags = ABSTRACT
```

You only need to include lines that you've changed from the parent class. It inherits everything else.

File: orbstation/code/antagonists/traitor/items/intergrated_chainsaw.dm [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	var/obj/item/bodypart/part
+	new /obj/item/chainsaw(get_turf(src))
+	if(iscarbon(loc))
+		var/mob/living/carbon/holder = loc
+		var/index = holder.get_held_index_of_item(src)
+		if(index)
+			part = holder.hand_bodyparts[index]
+	. = ..()
+	if(part)
+		part.drop_limb()
```
Comment (lizardqueenlexi) [inline orbstation/code/antagonists/traitor/items/intergrated_chainsaw.dm:59]:
```suggestion

```

Remove all this, it's the same as the parent.

**Review by lizardqueenlexi — CHANGES_REQUESTED** [review]

File: orbstation/code/antagonists/traitor/items/intergrated_chainsaw.dm (hunk 1 above)
Comment (Enbyy) [reply orbstation/code/antagonists/traitor/items/intergrated_chainsaw.dm:43]: > You only need to include lines that you've changed from the parent class. It inherits everything else.

oh fuck that's good to know thanks

**Review by lizardqueenlexi — APPROVED** [review]
I'd like it if these received unique sprites in the near future to make it more obvious that this is something distinct from the hand-replacing chainsaw. However, I think this is complete enough to put in the game.

**Pull request merged**